File: Natom.AccessMonitor.WebApp.Clientes.Backend/DTO/Background/ResumeDTO.cs
﻿using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Natom.AccessMonitor.WebApp.Clientes.Backend.DTO.Background
{
    public class ResumeDTO
    {
        [JsonProperty("current_year")]
        public int CurrentYear { get; set; }

        [JsonProperty("organization")]
        public OrganizationDTO Organization { get; set; }

        [JsonProperty("unassigned_devices")]
        public List<string> UnassignedDevices { get; set; }

        [JsonProperty("places_without_hours")]
        public List<string> PlacesWithoutHours { get; set; }

        [JsonProperty("syncs_times")]
        public List<SyncTimeDTO> SyncsTimes { get; set; }
    }
}

File: Natom.AccessMonitor.WebApp.Clientes.Frontend/src/app/classes/dto/background/resume.dto.ts
import { OrganizationDTO } from "./organization.dto";
import { SyncTimeDTO } from "./sync-time.dto";

export class ResumeDTO {
    public current_year: number;
    public organization: OrganizationDTO;
    public unassigned_devices: Array<string>;
    public places_without_hours: Array<string>;
    public syncs_times: Array<SyncTimeDTO>;
}